app/src/main/java/org/schabi/newpipe/util/AndroidTvUtils.java

```diff
@@ -1,22 +1,55 @@
 package org.schabi.newpipe.util;
 
-import android.annotation.SuppressLint;
+import android.app.UiModeManager;
+import android.content.Context;
 import android.content.pm.PackageManager;
-
+import android.content.res.Configuration;
+import android.os.BatteryManager;
+import android.os.Build;
 import android.view.KeyEvent;
+
 import org.schabi.newpipe.App;
 
+import static android.content.Context.BATTERY_SERVICE;
+import static android.content.Context.UI_MODE_SERVICE;
+
 public final class AndroidTvUtils {
+
     private static final String AMAZON_FEATURE_FIRE_TV = "amazon.hardware.fire_tv";
+    private static Boolean isTV = null;
```

Comment:
Please use the primitive type `boolean`

Reply:
No??
How do you know if it's the first time you check `isTv()` and have to do all the calculations? For the first time, object value is null, but primitive would be false, how do you difference unitialized `false` from `false` _real_ value with primitive?

I don't see any other proper way to acheive this https://github.com/TeamNewPipe/NewPipe/pull/3483#issuecomment-619447980 than using Boolean object and its null value, but you tell me.

Reply:
Oh, you're right, I forgot about that